### PR TITLE
test: [M3-7189] - Add assertions for backup prices in Linode Create flow

### PR DESCRIPTION
**Pull request by cliu-akamai**
## Description 📝
Add new cypress assertions for backup prices in Linode create flow.

## Major Changes 🔄
- Add new cypress assertions to check backup prices when creating a Linode.

## How to test 🧪
```
yarn cy:run -s "cypress/e2e/core/linodes/create-linode.spec.ts"
```

### CODE REVIEW

**Review by abailly-akamai — COMMENTED** [review]
Screenshot 2023-10-23 at 09 46 59

Screenshot 2023-10-23 at 09 44 22

Am getting consistent failures on the "creates a linode via cli" step finding a region Id

File: packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts
Comment (mjac0bs) [inline packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts:1]: Could we update the comment block above the test that `shows DC-specific pricing information during create flow` to note that backups are covered too? e.g.
```
/*
   * - Confirms DC-specific pricing UI flow works as expected during Linode creation.
   * - Confirms that pricing notice is shown in "Region" section.
   * - Confirms that notice is shown when selecting a region with a different price structure.
   * - Confirms that backups pricing is correct when selecting a region with a different price structure. 
   */
```

**Review by mjac0bs — APPROVED** [review]
Thanks, @cliu-akamai - this passed consistently for me and I confirmed that I saw this test check for the expected DC-specific backup prices in the Addons and Summary sections. 

Screenshot 2023-10-26 at 10 11 20 AM

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts
Comment (cliu-akamai) [reply packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts:1]: Sure

**Comment by mjac0bs** [comment]
Thanks for addressing that small bit of feedback, @cliu-akamai! I reran the tests on this this morning because there were a bunch of failures on the run for some reason. The last finished run still has a few failed, unrelated tests for some reason (`rebuild-linode.spec.ts`,`access-keys.smoke.spec.ts `, `object-storage.smoke.spec.ts `) but these all pass fine locally. These changes should be all set to merge. 🚢

**Review by jdamore-linode — APPROVED** [review]
Nice work! Thanks @cliu-akamai

**Pull request merged**